frontend/src/app/modules/login/log-in-page/log-in-page.component.ts

```diff
@@ -76,28 +66,48 @@ export class LogInPageComponent implements OnInit {
     }
 
     public signInWithGitHub() {
-        this.authService.signInWithGitHub().subscribe(
-            (user: User) => {
-                this.router.navigate(['/main']);
-                this.toastrNotification.showSuccess(`${user.userName} was successfully signed in`);
-                // add email sender to user.email
-            },
-            (error) => {
-                this.toastrNotification.showError(error);
-            },
-        );
+        this.authService.signInWithGitHub()
+            .pipe(
+                catchError((error: string) => {
+                    if (!error.toLowerCase().includes(this.providerName)) {
+                        this.toastrNotification.showError(error);
+                    } else {
+                        console.error(error);
+                    }
+
+                    return of(undefined);
+                }),
```

Comment:
I think you should write a wrapper over that observable to avoid writing the same catch error block twice